### DISCUSSION TITLE
Using Unifi Protect Cloudkey Gen2/UNVR and Frigate+Unifi-Cam-Proxy to stream and store motion from different camera's

### DISCUSSION THREAD

**Discussion by EnigmaWebdesign** [blakeblackshear/frigate#6520] [Ask A Question]
Hi all,

So far I'm pretty impressed with Frigate. A the moment I have Unifi Cam Proxy, Frigate and MQTT all working in a docker enviroment.
I was running on a shuttle pc with an i5 but I migrated the installation to a AMD 7950X system. I'm trying to get around 51 camera's working. It's a mix of Mobotix (really old ones and some without RTSP) and Vivotek.

I was just wondering if anyone out there have also been testing with Unifi Protect Cloudkey Gen2 or UNVR and got it working with the following:
- Downloading recordings not working, download 0kb.
- Motion working with non-Unifi camera's.

- Smart detection is working too but as soon as I go to "Detections" I sometimes can view the recording, sometimes not and again download gives 0kb.

It would be really sweet if someone can give me pointers. 

Br,

Steve

**Comment by nicoleise** [comment]
Hi Steve, 

I tested both and unifi cam proxy was a very buggy experience at best for me. Eventually I ditched UniFi Protect because of how many workarounds it would need. I did like the playback UI though. 

The motion detection worked on one camera out of five. 3 of one type, 2 of a different type, all setup identically. One detected motion fairly reliably, none of the others did. The proxies were identical in config and were running in separate LXCs on a host with ample ressources. Why only one cam would work made no sense.

The download feature was the same, it would work on some cameras yet not on others. Like above, it made no sense that it would work so unreliably.

I never saw smart detections work, but I attributed that to UniFi Protect possibly not allowing the feature because my cams were registered in the app as G3. The stream would crash when I attempted to alter settings or view anything in relation to smart detections. 

Eventually I got fed up with Ubiquity's efforts to prevent the fixes and workarounds that others made to their solution (their app for example), while completely failing to address the problems and shortcomings themselves, especially those they had promised to address, and thus I simply gave up on it and use Frigate now. Haven't looked back, although I do miss that playback UI (something similar is coming for Frigate though, I believe). 

I hope the comment helps - perhaps not in the way you intended, but (to me) knowing if it's worth trying or not is also helpful. 

Kr, 
Nicolai

**Reply by EnigmaWebdesign** [reply]
Hi Nicolai,

Thank you for posting your experiences with Unifi in combination with Frigate.
Once you have Unifi up and running with the Unifi camera's there are a lot of things really nice to what Ubiquiti offers.
If you haven't checked out the app yet, that one is really cool. 

But to stay on topic, I am interested in getting 3rd party camera's working with Unifi Protect if possible. 
Frigate on its own has lots of cool features but it's missing a timeline-based playback UI and also a dedicated app of its own.
I think Frigate has lots of potential for sure and would really love to see how it grows.

I hope others would come forward with their experiences and maybe have some ideas on how this works.

2 years ago there was a Discord channel that started with a few people and some developers because Unifi Video was killed by Unifi.
They really rocked and in a couple of weeks, there was a docker version of a UNVR which was upgradable and could run of a Raspberry Pi with added storage. Really cool project. I hope something like this could happen with unifi-cam-proxy in combination with Frigate and/or motion too. 

Anyway, I'm still open to more feedback from the community. 

Cheers,

Steve
